src/components/hack_forms/HackathonOverviewForm.js

```diff
@@ -0,0 +1,216 @@
+          }
+          name="start-time"
+          label="Start Time"
+          inputVariant="outlined"
+          variant="inline"
+          autoOk
+          required
+        />
+      </div>
+      <div className={classes.row}>
+        <DatePicker
+          className={classes.datePicker}
+          value={props.overview.endDate}
+          onChange={date =>
+            props.setOverview({ ...props.overview, endDate: date })
+          }
+          name="end-date"
+          label="End Date"
+          inputVariant="outlined"
+          variant="inline"
+          autoOk
+          required
+        />
+        <TimePicker
+          className={classes.timePicker}
+          value={props.overview.endDate}
+          onChange={date =>
+            props.setOverview({ ...props.overview, endDate: date })
+          }
+          name="end-time"
+          label="End Time"
+          inputVariant="outlined"
+          variant="inline"
+          autoOk
+          required
+        />
+      </div>
+      <TextField
+        className={classes.textField}
+        value={props.overview.location}
+        onChange={event =>
+          props.setOverview({ ...props.overview, location: event.target.value })
+        }
+        variant="outlined"
+        required
+        fullWidth
+        id="location"
+        label="Location"
+        name="location"
+        margin="normal"
+      />
+      <TextField
+        className={classes.marginedTextField}
+        value={props.overview.maxRegistrants}
+        onChange={event => {
+          const value = parseInt(event.target.value);
+          // Ensure that we have a positive number of registrants
+          if (!isNaN(value) && value > 0) {
+            props.setOverview({
+              ...props.overview,
+              maxRegistrants: value
+            });
+          }
+        }}
+        variant="outlined"
+        required
+        fullWidth
+        id="max-registrants"
+        label="Max Registrants"
+        name="max-registrants"
+        margin="normal"
+      />
+      <div className={classes.row}>
+        <DatePicker
+          className={classes.datePicker}
+          value={props.overview.regDeadline}
+          onChange={date =>
+            props.setOverview({ ...props.overview, regDeadline: date })
+          }
+          name="registration-deadline-date"
+          label="Registration Deadline Date"
+          inputVariant="outlined"
+          variant="inline"
+          autoOk
+          required
+        />
+        <TimePicker
+          className={classes.timePicker}
+          value={props.overview.regDeadline}
+          onChange={date =>
+            props.setOverview({ ...props.overview, regDeadline: date })
+          }
+          name="registration-deadline-time"
+          label="Registration Deadline Time"
+          inputVariant="outlined"
+          variant="inline"
+          autoOk
+          required
+        />
```

Comment:
Logic looks good, but this is huge! Is it possible to break it down at all? `CodeClimate` would have a conniption over this haha. (Not that we're going to use it)

Reply:
Ha!! Yeah, uh, definitely can break it down. In the next PR for previewing a hackathon, I ended up refactoring this (date pickers only take 3 lines rather than nearly 30 in this version). If it's ok with you, I'll merge this (rather massive) code as-is and then it'll all get fixed in the next PR!